baybe/searchspace/discrete.py

```diff
@@ -255,16 +255,24 @@ def from_product(
         # Set defaults
         constraints = constraints or []
 
-        # Create a dataframe representing the experimental search space
-        exp_rep = parameter_cartesian_prod_to_df(parameters)
+        # Use Polars for Cartesian product of parameters and apply constraint
+        lazy_df = parameter_cartesian_prod_polars(parameters)
+        lazy_df = _apply_polars_constraint_filter(lazy_df, constraints)
+        pl_df = lazy_df.collect(streaming=True)
 
-        # Remove entries that violate parameter constraints
-        _apply_constraint_filter(exp_rep, constraints)
+        # Check if df is empty to prevent Polar's error
+        if pl_df.shape[0] == 0:
+            df = pd.DataFrame()
```

Comment:
why is this if/else block needed in the first place? `pl.DataFrame().to_pandas()` works for me without problems. Am I overlooking something?

Reply:
In this way you're first creating a Polars dataframe and then do a conversion.
Since you were concerned about possible performance overhead, I wanted to avoid it (however it's negligible here)

Reply:
That was not my point, I just wanted to demonstrate that converting an empty polars dataframe to pandas causes no problems. So why do you then have this if/else block in the first case? The first case simply covers the case that you have an empty dataframe, and I do not see the reason why it should not be possible to simply call `df = pl_df.to_pandas()` in both cases?

Reply:
I remember that there was a problem with empty Polars dataframes, but you're right, I no longer see related errors after running tests, so I remove the redundant check.

**TL;DR: Fixed**